rs/src/management/http_client.rs

```diff
@@ -533,10 +536,32 @@ pub struct TunnelClientBuilder {
 /// to get the client instance (or cast automatically).
 pub fn new_tunnel_management(user_agent: &str) -> TunnelClientBuilder {
     let pkg_version = PKG_VERSION.unwrap_or("unknown");
-    let full_user_agent = format!(
-        "{}{}{}",
-        user_agent, " Dev-Tunnels-Service-Rust-SDK/", pkg_version
+    let os = os_info::get();
+    let os_info = format!("{}: {} {}", "OS", os.os_type(), os.version());
+    let mut windows_partner_id = String::default();
+    #[cfg(windows)]
+    {
+        let hklm = RegKey::predef(HKEY_LOCAL_MACHINE);
+        let key = hklm.open_subkey("SOFTWARE\\Microsoft\\Windows365");
+        windows_partner_id = key
+            .expect("No value found for key")
+            .get_value::<String, _>("PartnerId")
+            .unwrap_or_default();
```

Comment:
once windows_partner_id is an option type, and if we can silently ignore the lack of a key, this can just be `window_partner_id = key.and_then(|k| k.get_value("PartnerId")).ok()`

Reply:
Thanks for suggesting this. It also prevents the program from breaking for a lack of key, which I agree it shouldn't do since this key won't exist for many users